packages/frontend/src/components/buy/BuyNear.js

```diff
@@ -144,13 +144,13 @@ const StyledContainer = styled(Container)`
 `;
 
 export function BuyNear({ match, location, history }) {
-    const { accountId } = useSelector(selectAccountId);
+    const accountId = useSelector(selectAccountId);
     const [moonPayAvailable, setMoonPayAvailable] = useState(null);
     const [signedMoonPayUrl, setSignedMoonPayUrl] = useState(null);
 
     useEffect(() => {
         checkMoonPay();
-    }, []);
+    }, [accountId]);
```

Comment:
Should we add `accountId` into the `if` statement on `L159` too so we make sure we don't show Moonpay as available if the `accountId` is slow to load for some reason?

Reply:
Added condition. I also propose to don't hide moonpay section, instead, add the loader to the button - thanks to that the content will not jump after accountId is available.

Reply:
Nice. While I suspect Moonpay is supported in most regions now, we probably want to handle the case where it's not. I suggest we keep the 'loading...' state until we've received a response from `isMoonpayAvailable`. If Moonpay is not supported, we can disabled the button and change the text to 'Not supported in your region'

cc @corwinharrell

Reply:
I see. Added.
![Screenshot 2021-12-01 at 21 59 07](https://user-images.githubusercontent.com/36210360/144314540-bf873db7-d2a5-414e-a8aa-4e9e7696605d.png)